virtual_ship/expedition/simulate_measurements.py

```diff
@@ -0,0 +1,104 @@
+"""simulate_measurements function."""
+
+from datetime import timedelta
+from pathlib import Path
+
+from ..instruments.adcp import simulate_adcp
+from ..instruments.argo_float import simulate_argo_floats
+from ..instruments.ctd import simulate_ctd
+from ..instruments.drifter import simulate_drifters
+from ..instruments.ship_underwater_st import simulate_ship_underwater_st
+from .input_data import InputData
+from .ship_config import ShipConfig
+from .simulate_schedule import MeasurementsToSimulate
+
+
+def simulate_measurements(
+    expedition_dir: str | Path,
+    ship_config: ShipConfig,
+    input_data: InputData,
+    measurements: MeasurementsToSimulate,
+) -> None:
+    """
+    Simulate measurements using parcels.
+
+    Saves everything the $expedition_dir/results.
+
+    :param expedition_dir: Base directory of the expedition.
+    :param ship_config: Ship configuration.
+    :param input_data: Input data for simulation.
+    :param measurements: The measurements to simulate.
+    :raises RuntimeError: In case fieldsets of configuration is not provided. Make sure to check this before calling this function.
+    """
+    if isinstance(expedition_dir, str):
+        expedition_dir = Path(expedition_dir)
+
+    if len(measurements.ship_underwater_sts) > 0:
+        print("Simulating onboard salinity and temperature measurements.")
+        if input_data.ship_underwater_st_fieldset is None:
+            raise RuntimeError("No fieldset for ship underwater ST provided.")
```

Comment:
This comes from the default fieldset? Is that clear with this error? Same for other measurements below

Reply:
I do not understand your question. The fieldsets must be available in the InputData object. If they are not loaded, this code cannot simulate